### ISSUE TITLE
[iOS] [Build] and [Publish] fail with `NETSDK1047: Assets file 'project.assets.json' doesn't have a target for 'net6.0-ios/ios-arm64'`

### ISSUE THREAD

**Issue by xperiandri** [xamarin/xamarin-macios#14753]
### Steps to Reproduce

1. Create a new MAUI app and restore packages
2. Execute either
`dotnet build --no-restore -f net6.0-ios -r ios-arm64 -c Release /p:ArchiveOnBuild=true --self-contained`
or
`dotnet publish --no-restore -f net6.0-ios -r ios-arm64 -c Release --self-contained`

### Expected Behavior
Binaries saved to `bin/Release/net6.0-ios`
ipa saved to `bin/Release/net6.0-ios/ios-arm64/publish`

### Actual Behavior
```
xperiandri@Andrii-Mac Client.Mobile % dotnet publish --no-restore -f net6.0-ios -r ios-arm64 -c Release
Microsoft (R) Build Engine version 17.2.0-preview-22126-01+40f49cd8d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  Detected signing identity:
          
    Code Signing Key: 
    Provisioning Profile: 
    Bundle Id: 
    App Id: 
/usr/local/share/dotnet/sdk/6.0.300-preview.22154.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1047: Assets file '/Users/xperiandri/Projects/ITM/Client/Client.Mobile/obj/project.assets.json' doesn't have a target for 'net6.0-ios/ios-arm64'. Ensure that restore has run and that you have included 'net6.0-ios' in the TargetFrameworks for your project. You may also need to include 'ios-arm64' in your project's RuntimeIdentifiers. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]

Build FAILED.

/usr/local/share/dotnet/sdk/6.0.300-preview.22154.4/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1047: Assets file '/Users/xperiandri/Projects/ITM/Client/Client.Mobile/obj/project.assets.json' doesn't have a target for 'net6.0-ios/ios-arm64'. Ensure that restore has run and that you have included 'net6.0-ios' in the TargetFrameworks for your project. You may also need to include 'ios-arm64' in your project's RuntimeIdentifiers. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
    0 Warning(s)
    1 Error(s)
```

### Environment
macOS 12.3.1
```
Visual Studio Community 2022 for Mac Preview
Version 17.0 Preview (17.0 build 8567)
Installation UUID: b9858b41-932e-430f-9a57-7b2f9675cc3a

Runtime
.NET 6.0.3 (64-bit)
Architecture: X64

Xamarin Designer
Version: 17.1.5.85
Hash: bd4553834
Branch: remotes/origin/d17-0-vsmac
Build date: 2022-04-07 20:47:56 UTC

Roslyn (Language Service)
4.1.0-3.22075.3+592501cbb9c9394072a245c15b3458ff88155d85

NuGet
Version: 6.0.0.262

.NET SDK (x64)
SDK: /usr/local/share/dotnet/sdk/6.0.300-preview.22154.4/Sdks
SDK Versions:
	6.0.300-preview.22154.4
	6.0.201
	6.0.101
	5.0.407
	5.0.406
	3.1.418
	3.1.417
MSBuild SDKs: /usr/local/share/dotnet/sdk/6.0.300-preview.22154.4/Sdks

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.3
	6.0.2
	5.0.16
	5.0.15
	3.1.24
	3.1.23

Xamarin.Profiler
Version: 1.8.0.1
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin.Android
Not Installed

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.10
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: Not Found

Android SDK Manager
Version: 17.2.0.31
Hash: d341462
Branch: remotes/origin/dev/mag/d17-2~1
Build date: 2022-04-07 20:48:01 UTC

Android Device Manager
Version: 0.0.0.1110
Hash: 6cb643c
Branch: remotes/origin/d17-2
Build date: 2022-04-07 20:48:01 UTC

Apple Developer Tools
Xcode 13.3 (20102)
Build 13E113

Xamarin.Mac
Version: 8.8.0.3 (Visual Studio Community)
Hash: eb7f8ed35
Branch: xcode13.3
Build date: 2022-03-30 03:16:09-0400

Xamarin.iOS
Version: 15.8.0.3 (Visual Studio Community)
Hash: eb7f8ed35
Branch: xcode13.3
Build date: 2022-03-30 03:16:09-0400

Build Information
Release ID: 1700008567
Git revision: 95e0843576bf05769cbede768cc99d36f75ac629
Build date: 2022-04-07 20:45:45+00
Build branch: release-17.0
Build lane: release-17.0

Operating System
Mac OS X 12.3.1
Darwin 21.4.0 Darwin Kernel Version 21.4.0
    Fri Mar 18 00:45:05 PDT 2022
    root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64
```

**Comment by rolfbjarne** [comment]
This works if you pass the runtime identifier when restoring:

```shell
$ dotnet restore /p:RuntimeIdentifier=ios-arm64
...
$ dotnet build ...
```

The restore needs to know which runtime identifier will be used later at build time, and the default is the simulator runtime identifier (`iossimulator-x64`), which means that's what will be restored. If you want to build using any other runtime identifier, you need to pass it to restore.

**Comment by xperiandri** [comment]
```
dotnet restore /p:RuntimeIdentifier=ios-arm64
  Determining projects to restore...
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/12.3.100-rc.1.125/targets/Xamarin.Shared.Sdk.targets(1732,3): error : The RuntimeIdentifier 'ios-arm64' is invalid. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
```

**Comment by xperiandri** [comment]
`The RuntimeIdentifier 'ios-arm64' is invalid.` what is wrong?

**Comment by xperiandri** [comment]
This works!

**Comment by xperiandri** [comment]
Now it is going better but still error
```

**Comment by xperiandri** [comment]
So `<MtouchUseLlvm>False</MtouchUseLlvm>` fixes the issue for Release
@rolfbjarne would you like me to create a new issue for `Failed to AOT compile FSharp.Core.dll and SharpPlus.dll with LLVM?

**Comment by xperiandri** [comment]
`publish` works too but it has extra slash in the path
```
dotnet publish -f net6.0-ios -r ios-arm64 -c Release --no-restore --self-contained 
Microsoft (R) Build Engine version 17.2.0-preview-22178-04+ce7523296 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  Detected signing identity:
          
    Code Signing Key: 
    Provisioning Profile: 
    Bundle Id: 
    App Id: 
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  Transmission.FSharp -> /Users/xperiandri/Projects/ITM/Shared/FSharp/bin/Release/netstandard2.0/Transmission.FSharp.dll
  Transmission.Domain -> /Users/xperiandri/Projects/ITM/Shared/Domain/bin/Release/netstandard2.0/Transmission.Domain.dll
  Transmission.Client -> /Users/xperiandri/Projects/ITM/Client/Client/bin/Release/netstandard2.0/Transmission.Client.Programs.dll
  Uno.UI is using Roslyn generators
  Transmission.Client.Mobile -> /Users/xperiandri/Projects/ITM/Client/Client.Mobile/bin/Release/net6.0-ios/ios-arm64/Transmission.Client.Mobile.dll
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.100-rc.1.125/targets/Xamarin.Shared.Sdk.targets(1432,3): warning : The file '/Users/xperiandri/.nuget/packages/fsharp.core/6.0.3/contentFiles/any/netstandard2.1/FSharp.Core.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.100-rc.1.125/targets/Xamarin.Shared.Sdk.targets(1432,3): warning : The file '/Users/xperiandri/Projects/ITM/Client/Client/bin/Release/netstandard2.0/Transmission.Client.Programs.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.100-rc.1.125/targets/Xamarin.Shared.Sdk.targets(1432,3): warning : The file '/Users/xperiandri/Projects/ITM/Shared/Domain/bin/Release/netstandard2.0/Transmission.Domain.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.100-rc.1.125/targets/Xamarin.Shared.Sdk.targets(1432,3): warning : The file '/Users/xperiandri/Projects/ITM/Shared/FSharp/bin/Release/netstandard2.0/Transmission.FSharp.xml' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle. [/Users/xperiandri/Projects/ITM/Client/Client.Mobile/Transmission.Client.Mobile.csproj]
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Created the package: bin/Release/net6.0-ios/ios-arm64/publish//Transmission.Client.Mobile.ipa
```

**Comment by rolfbjarne** [comment]
> So `<MtouchUseLlvm>False</MtouchUseLlvm>` fixes the issue for Release
> @rolfbjarne would you like me to create a new issue for `Failed to AOT compile FSharp.Core.dll and SharpPlus.dll with LLVM?

Yes, please, that would be great. Please include a project we can use to reproduce the error.



> `publish` works too but it has extra slash in the path

That should be fixed with the fix for https://github.com/xamarin/xamarin-macios/issues/14751.

**Comment by rolfbjarne** [comment]
> > `publish` works too but it has extra slash in the path
> 
> That should be fixed with the fix for #14751.

It's also harmless to have double slashes in paths, so it shouldn't cause any problems.
